### ISSUE TITLE
A way to check queue existence

### ISSUE THREAD

**Issue by blanchma** [alanxz/rabbitmq-c#520]
Is a way to verify the existence of a queue without a declare? The issue with declare is requires to know the params of the creation. I just need to know if the queue was already created.

**Comment by manchicken** [comment]
I don't think that this is how AMQP works. https://www.rabbitmq.com/amqp-0-9-1-reference.html#class.queue is the spec for everything you can do with a queue.

What you're describing is part of the protocol. Depending on your use-case, you may want to use an error call to detect the queue. You could try declaring the queue with parameters you _know_ are different, and make it auto-delete. If you get an error, you know the queue exists already. If you don't get an error, you know the queue is missing and then your test queue will be automatically deleted.

**Comment by ostreamBaba** [comment]
`bool checkQueueExist(const std::string queue)
{
    amqp_queue_declare(m_conn, m_channel, amqp_cstring_bytes(queue.c_str()), 1, 0, 0, 0, amqp_empty_table);
    if (!die_on_amqp_error_resp(amqp_get_rpc_reply(m_conn), "Declaring queue", m_error_msg)) {
        return false;
    }
    return true;
}`

**Comment by ostreamBaba** [comment]
`
bool MasterRabbitmqRpcServer::checkQueueExist(const std::string queue)
{
    amqp_queue_declare(m_conn, m_channel, amqp_cstring_bytes(queue.c_str()), 1, 0, 0, 0, amqp_empty_table);
    if (!die_on_amqp_error_resp(amqp_get_rpc_reply(m_conn), "Declaring queue", m_error_msg)) {
        return false;
    }
    return true;
}
`

**Comment by ostreamBaba** [comment]
```
bool checkQueueExist(const std::string queue)
{
    amqp_queue_declare(m_conn, m_channel, amqp_cstring_bytes(queue.c_str()), 1, 0, 0, 0, amqp_empty_table);
    if (!die_on_amqp_error_resp(amqp_get_rpc_reply(m_conn), "Declaring queue", m_error_msg)) {
        return false;
    }
    return true;
}
```
